### PR TITLE
feat: add regular expression category and resources

### PR DESCRIPTION
**Pull request by savioxavier**
Hello there! 👋

This pull request adds a new category to the list - Regular Expressions

I personally believe Regular Expressions (or regex, or regexp, call it whatever you will) are incredibly helpful for any programmer, so I've added this category.

I have also taken the liberty of adding a few relevant resources, including tools and short courses, under the same topic

Cheers!

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -916,6 +920,14 @@ Throughout this list you'll see next to each resource and emoji. Here's what eac
 
 ---
 
+## Regular Expressions
```
Comment (shahednasser) [inline README.md:923]: Please add before this the "Back to Index" link. Refer to the [contribution guidelines](https://github.com/shahednasser/awesome-resources/blob/master/CONTRIBUTING.md) for more details.

**Review by shahednasser — CHANGES_REQUESTED** [review]
Thank you for your contribution! I added some comments below, but in addition I can see you made changes to all of the table of content. Can you just add a new link for Regular Expressions and revert all changes?

**Comment by savioxavier** [comment]
No problemo, I'll fix those changes. It looks like my markdown formatter accidently updated the table of contents though.

**Comment by savioxavier** [comment]
Unfortunately, I think I messed up the `git revert` (rookie mistake) and although I have fixed it in another remote branch, it's not possible to change the target branch now. I may need to open another pull request. I apologize for the inconvenience.

**Comment by savioxavier** [comment]
There we go. Please check out the new PR and let me know if there' anything left to be done. Thanks!

**Pull request closed without merging**